zerver/lib/message.py

```diff
@@ -1117,9 +1117,21 @@ def extract_unread_data_from_um_rows(
     topic_visibility_policy_checker = build_topic_visibility_policy_checker(user_profile)
 
     def is_row_muted(stream_id: int, recipient_id: int, topic: str) -> bool:
-        if stream_id in muted_stream_ids:
+        stream_muted = stream_id in muted_stream_ids
```

Comment:
Side note: We should probably change `get_muted_stream_ids` to return a `Set`; we're doing quadratic work here for no reason.

Reply:
Added a prep commit. -- 3rd commit